chafa: add page (#8413)
tldr-pages/tldr: pages/common/chafa.md

# chafa

> Image printing in the terminal.
> See also: `catimg` and `pixterm`.
> More information: <https://hpjansson.org/chafa>.

> - Print an image or animated GIF to the terminal:

Reviewer: All the tldr examples are for the terminal, should the user use ctrl + C or q to exit?
```suggestion
- Print an image or animated GIF:
```

Author: It's creating an image out of text characters inside the terminal. And I generally think of "print to the terminal" as a sort of phrase, since just using "print" might make you think of a GUI application or an actual printer.

I already pretty much said this in the description though, so I'd understand if it seems redundant.

> - Render an image or animated GIF directly in the terminal:

Reviewer 2: ```suggestion
- Render an image or a GIF directly in the terminal:
```

Author: I was thinking it might be an important distinction. `pixterm`, for example, can only render static GIFs. But GIF is synonymous with animation for most people, so maybe that isn't so useful. I might just omit the GIF mention entirely.

```suggestion
- Render an image directly in the terminal:
```

Author: Yeah, I'm just removing it.

Reviewer 2: to me, a gif is a collection of pictures put together (just described a video :smile: ) so in my definition, it is an animation.

Author: Maybe I'm just being too particular then.

> `chafa {{path/to/input}}`

Reviewer: We prefer `path/to/file`
```suggestion
`chafa {{path/to/file}}`
```

Author: Got it. I'll update that here and in my `pixterm` and `catimg` PRs.

Author: Updated.

> - Use 24-bit [c]olor:

Reviewer: ```suggestion
- Print an image using 24-bit [c]olor:
```

Author: I think it gets a little dry to say "print" over and over. I already specified that it prints to the terminal in the first example. And since its primary function is to generate images and print them to the terminal, I think "use" should be good enough.

Reviewer: The example descriptions shouldn't be described using the context. Make them independent

Author: Hm. I was under the impression that the format is useful for both showing independent examples and giving a brief tutorial as a whole. This is even somewhat supported by the guides, where it says to start with simple examples before going to more complex ones. Maybe this particular point isn't a good example of that, but is it really better to use the same verbage over and over, when the primary use of the tool has already been demonstrated?

`chafa -c full {{path/to/input}}`

> - Use dithering to improve output with small color palettes:

Reviewer: ```suggestion
- Print an image using dithering to improve output with small color palettes:
```

`chafa -c 16 --dither ordered {{path/to/input}}`

> - Generate output like a low-resolution raster image:

Reviewer: Why `generate` and not `print` here?

Author: Printing to terminal is just "spitting out" what it's generated. In the last two examples, I think it's a little more specific to say something like "generate" instead of the verb "use" I'd used before that. So basically, nothing about the printing process is being changed, whereas the process of generation *is*. And it's not simply using a one feature; the whole process is *generating* something similar to a raster image. I hope my reasoning makes sense.

> - Render an image in the terminal to appear pixelated:

Reviewer 2: ```suggestion
- Render an image, making it appear pixelated:
```

`chafa --symbols vhalf {{path/to/file}}`

> - Render a monochrome image with only Braille characters:

Reviewer 2: ```suggestion
- Render a monochrome image with only braille characters:
```

Not sure whether this should be capitalized, from a little research, I think it shouldn't be.

Author: Nice. I hadn't realized.
